csrc/multidevice/cuda_p2p.cpp

```diff
@@ -0,0 +1,70 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <cuda_utils.h>
+#include <multidevice/cuda_p2p.h>
+
+namespace nvfuser {
+
+namespace getZcopy {
+
+void RecvPost(const P2pIpcHandle& ipc_handles, int64_t count, CUstream stream) {
+  // wait for sender to be ready
+  NVFUSER_CUDA_SAFE_CALL(cuStreamWaitValue32(
+      stream,
+      reinterpret_cast<CUdeviceptr>(ipc_handles.local().semaphore()),
+      (cuuint32_t)(IpcSemaphore::kInUse),
+      CU_STREAM_WAIT_VALUE_EQ));
+  // RDMA get the data from the sender
+  NVFUSER_CUDA_RT_SAFE_CALL(cudaMemcpyAsync(
+      ipc_handles.local().ptr(),
+      ipc_handles.peer().ptr(),
+      count,
+      cudaMemcpyDeviceToDevice,
+      stream));
+  // Signals completion to self
+  NVFUSER_CUDA_SAFE_CALL(cuStreamWriteValue32(
+      stream,
+      reinterpret_cast<CUdeviceptr>(ipc_handles.local().semaphore()),
+      (cuuint32_t)(IpcSemaphore::kReady),
+      CU_STREAM_WRITE_VALUE_DEFAULT));
+  // Signals completion to receiver
```

Comment:
```suggestion
  // Signals completion to sender
```
Is that right? I thought for recvPost local==receiver and peer==sender.

Reply:
you're right, that's a typo